plugin/storage/es/factory.go

```diff
@@ -36,16 +36,15 @@ import (
 )
 
 const (
-	primaryNamespace = "es"
-	archiveNamespace = "es-archive"
+	PrimaryNamespace = "es"
+	ArchiveNamespace = "es-archive"
 )
 
 var ( // interface comformance checks
-	_ storage.Factory        = (*Factory)(nil)
-	_ storage.ArchiveFactory = (*Factory)(nil)
```

Comment:
need to be very careful about removing ArchiveFactory interface, because query service uses it via runtime cast, so unless we have integration tests (one many of them disable archive tests as I recall) you can introduce a breaking change https://github.com/jaegertracing/jaeger/blob/772d84c3976c8ee688043f0a052d7198de30053e/cmd/query/app/querysvc/query_service.go#L131

Reply:
@yurishkuro yep - noted. I was thinking to avoid breaking changes we can remove the ArchiveFactory within this PR.

Reply:
https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/integration/elasticsearch_test.go#L165 - ES doesn't skip the archive test